### PR TITLE
Adding a new role for marketplace apps

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Adds the new marketplace app permission and role to the enterprise table
Adds the role as assignable to service tokens
Adds the role as a pre-requisite for the Snowflake Native App

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/docs/dbt-cloud-apis/service-tokens.md [hunk 1]
```diff
@@ -69,6 +69,9 @@ Security Admin service tokens have certain account-level permissions.  For more
 **Billing Admin** <br/>
 Billing Admin service tokens have certain account-level permissions.  For more on these permissions, see [Billing Admin](/docs/cloud/manage-access/enterprise-permissions#billing-admin).
 
+**Marketplace app** <br/>
```
Comment (matthewshaver) [inline website/docs/docs/dbt-cloud-apis/service-tokens.md]:
```suggestion
**Manage marketplace apps** <br/>
```

File: website/docs/docs/dbt-cloud-apis/service-tokens.md [hunk 2]
```diff
@@ -69,6 +69,9 @@ Security Admin service tokens have certain account-level permissions.  For more
 **Billing Admin** <br/>
 Billing Admin service tokens have certain account-level permissions.  For more on these permissions, see [Billing Admin](/docs/cloud/manage-access/enterprise-permissions#billing-admin).
 
+**Manage marketplace apps** <br/>
+Used only for apps on the marketplace (for example, the [Snowflake Native app](/docs/cloud-integrations/snowflake-native-app)).
```
Comment (matthewshaver) [inline website/docs/docs/dbt-cloud-apis/service-tokens.md]:
```suggestion
Used only for service tokens assigned to marketplace apps (for example, the [Snowflake Native app](/docs/cloud-integrations/snowflake-native-app)). 
```

File: website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ The following are the prerequisites for dbt Cloud and Snowflake.
 
 - You must have a dbt Cloud account on the Enterprise plan that's in an AWS Region or Azure region. If you don't already have one, please [contact us](mailto:sales_snowflake_marketplace@dbtlabs.com) to get started.
     - Currently, Semantic Layer is unavailable for Azure ST instances and the **Ask dbt** chatbot will not function in the dbt Snowflake Native App without it. 
-- Your dbt Cloud account must have permission to create a [service token](/docs/dbt-cloud-apis/service-tokens). For details, refer to [Enterprise permissions](/docs/cloud/manage-access/enterprise-permissions).
+- Your dbt Cloud account must have permission to create a [service token](/docs/dbt-cloud-apis/service-tokens) assigned the `Manage marketplace apps` role. For details, refer to [Enterprise permissions](/docs/cloud/manage-access/enterprise-permissions).
```
Comment (runleonarun) [inline website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md]: Lower down on this page, we **also** say to grant Job Admin, Metadata, Semantic Layer Only. IS there a reason we're just mentioning this one or should we mention these as well?

https://github.com/dbt-labs/docs.getdbt.com/pull/5815/files#diff-770944d95df777c9ab980c64ab039e22058f83e3eaa322e78fca7429f397c128L71-L73

**Review by runleonarun — APPROVED** [review]
@matthewshaver This looks great! Just one question.

File: website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md (hunk 3 above)
Comment (matthewshaver) [inline website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md]:
```suggestion
- Your dbt Cloud account must have permission to create a [service token](/docs/dbt-cloud-apis/service-tokens). For details, refer to [Enterprise permissions](/docs/cloud/manage-access/enterprise-permissions).
```

Comment (matthewshaver) [reply website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md]: That's a good catch. That shouldn't be in that spot, reverting it to the way it was and relying on the below info only!

File: website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md [hunk 4]
```diff
@@ -68,6 +68,7 @@ Configure dbt Cloud and Snowflake Cortex to power the **Ask dbt** chatbot.
 Collect three pieces of information from dbt Cloud to set up the application.
```
Comment (runleonarun) [inline website/docs/docs/cloud-integrations/set-up-snowflake-native-app.md]: You could also say "four?"
```suggestion
Collect the following pieces of information from dbt Cloud to set up the application. 
```

**Review by runleonarun — APPROVED** [review]
Looks good, just one small suggesiton!

**Pull request merged**